functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/pages/appadmin/KeyStoragePage.groovy

```diff
@@ -0,0 +1,123 @@
+package org.rundeck.tests.functional.selenium.pages.appadmin
+
+import groovy.transform.CompileStatic
+import org.openqa.selenium.By
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.Select
+import org.rundeck.tests.functional.selenium.pages.BasePage
+import org.rundeck.util.container.SeleniumContext
+
+/**
+ * Key Storage page
+ */
+@CompileStatic
+class KeyStoragePage extends BasePage {
+
+    By addUploadKey = By.xpath("//button[contains(.,'Key')] | //*[contains(@href,'uploadkey')]")
+    By uploadKeyType = By.name("uploadKeyType")
+    By uploadPassword = By.id("uploadpasswordfield")
+    By uploadPath = By.id("uploadResourcePath2")
+    By uploadResourceName = By.id("uploadResourceName2")
+    By save = By.xpath("//button[contains(.,'Save')] | //button[@class='btn btn-cta'] | //input[contains(@type, 'submit')]")
+    By overBy = By.xpath("//button[contains(.,'Overwrite')] | //a[contains(@data-bind, 'actionUploadModify')]")
+
+    String loadPath = "/menu/storage"
+
+    KeyStoragePage(final SeleniumContext context) {
+        super(context)
+    }
+
+    void validatePage() {
+        if (!driver.currentUrl.contains(loadPath)) {
+            throw new IllegalStateException("Not on key storage page: " + driver.currentUrl)
+        }
+    }
+
+    def goToKey(String name, String storagePath) {
+        By storageBy = By.xpath("//*[@class=\"action\"]//*[contains(.,'$storagePath')]")
+        waitForElementVisible storageBy
+        def storageEl = el storageBy
+        storageEl.click()
+
+        By nameBy = By.xpath("//*[@class=\"action\"]//*[contains(.,'$name')]")
+        waitForElementVisible nameBy
+        def nameEl = el nameBy
+        nameEl.click()
+    }
+
+    void addPasswordType(String keyValue, String storagePath, String name) {
+        def select = new Select(uploadKeyTypeField)
+        select.selectByValue("password")
+        uploadPasswordField.sendKeys(keyValue)
+        uploadPathField.sendKeys(storagePath)
+        uploadResourceNameField.sendKeys(name)
+        sleep 2000
```

Comment:
why is this sleep necessary?

Reply:
Is replaced with `waitForElementToBeClickable`